pkg/kubelet/kubelet_test.go

```diff
@@ -1407,6 +1407,109 @@ func TestCreateMirrorPod(t *testing.T) {
 	}
 }
 
+func TestPodStartDurationCountMetric(t *testing.T) {
+	metrics.Register()
+
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	// Pod with a pending status.
+	pod1 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: "uid1",
+			Annotations: map[string]string{
+				kubetypes.ConfigFirstSeenAnnotationKey: kubetypes.NewTimestamp().GetString(),
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container1"},
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+		},
+	}
+	// Pod with a running status.
+	pod2 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: "uid2",
+			Annotations: map[string]string{
+				kubetypes.ConfigFirstSeenAnnotationKey: kubetypes.NewTimestamp().GetString(),
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container2"},
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+	}
+	// Pod1 container's status
+	pod1Status := &kubecontainer.PodStatus{
+		ID: pod1.UID,
+		ContainerStatuses: []*kubecontainer.Status{
+			{
+				Name:  "container1",
+				State: kubecontainer.ContainerStateCreated,
+			},
```

Comment:
why does the container status here need to be populated? Can we instead just pass in `&kubecontainer.PodStatus{ID: pod.UID}` directly into the `SyncPod` call?